packages/microsoft-security-adapter/src/definitions/fetch/intune/utils/platform_script.ts

```diff
@@ -104,11 +111,87 @@ export const setWindowsScriptValueAsStaticFile: AdjustFunctionMergeAndTransform
     value: {
       ..._.omit(value, SCRIPT_CONTENT_RECURSE_INTO_FIELD_NAME),
       [SCRIPT_CONTENT_FIELD_NAME]: createStaticFileFromScript({
-        typeName: PLATFORM_SCRIPT_WINDOWS_TYPE_NAME,
-        uniquePath: value[NAME_ID_FIELD.fieldName],
-        fileExtension: 'ps1',
+        typeName,
+        fullName: value[NAME_ID_FIELD.fieldName],
+        fileName: value.fileName,
         content: _.get(scriptContent[0], SCRIPT_CONTENT_FIELD_NAME),
       }),
     },
   }
 }
+/**
+ * Creates a fetch definition for fetching macOS or windows platform scripts.
+ * Includes fetching the script content separately for each script, and transforming it into a static file.
+ */
+export const createPlatformScriptFetchDefinition = ({
+  typeName,
+  path,
+  platform,
+}: {
+  typeName: string
+  path: EndpointPath
+  platform: 'Windows' | 'MacOS'
+}): FetchCustomizations => ({
+  [typeName]: {
+    requests: [
+      {
+        endpoint: {
+          path,
+          queryArgs: {
+            $expand: 'assignments',
+          },
+        },
+        transformation: {
+          ...DEFAULT_TRANSFORMATION,
+          omit: [ASSIGNMENTS_ODATA_CONTEXT],
+        },
+      },
+    ],
+    resource: {
+      directFetch: true,
+      recurseInto: {
+        // For some reason the script content is returned as null when listing the scripts,
+        // so we need to fetch it separately by making another request for each script
```

Comment:
we should try to keep comments relatively concise 🙏 
```suggestion
   // additional request to get script's content
```